core/src/main/java/org/apache/accumulo/core/rpc/clients/TServerClient.java

```diff
@@ -72,11 +72,23 @@ default Pair<String,C> getThriftServerConnection(Logger LOG, ThriftClientTypes<C
     }
     ZooCache zc = context.getZooCache();
     for (String serverPath : serverPaths) {
-      for (String server : zc.getChildren(serverPath)) {
-        var zLocPath = ServiceLock.path(serverPath + "/" + server);
-        zc.getLockData(zLocPath).map(sld -> sld.getAddress(service))
-            .map(address -> new ThriftTransportKey(address, rpcTimeout, context))
-            .ifPresent(servers::add);
+      if (!serverPath.endsWith(Constants.ZCOMPACTORS)) {
+        for (String server : zc.getChildren(serverPath)) {
+          var zLocPath = ServiceLock.path(serverPath + "/" + server);
+          zc.getLockData(zLocPath).map(sld -> sld.getAddress(service))
+              .map(address -> new ThriftTransportKey(address, rpcTimeout, context))
+              .ifPresent(servers::add);
+        }
+      } else {
```

Comment:
If this lookup is only used for `Constants.ZCOMPACTORS` then why not just check for that vs having it be the default `else` route?

Reply:
I pushed the change, but GitHub is having an issue right now with Pull Requests, so it's not showing up.